src/quo2/components/wallet/transaction_progress/view.cljs

```diff
@@ -0,0 +1,295 @@
+
+(defn calculate-box-state
+  [networkState counter index]
+  (cond
+    (and (= networkState :sending) (>= counter index) (< index 3))                 :confirmed
+    (and (= networkState :confirmed) (>= counter index) (< index 5))               :confirmed
+    (and (= networkState :finalising) (>= counter index) (< index 5))              :confirmed
+    (and (= networkState :finalising) (>= counter index) (> index 4) (< index 20)) :finalized
+    (and (= networkState :finalized) (>= counter index) (< index 5))               :confirmed
+    (and (= networkState :finalized) (>= counter index) (> index 4))               :finalized
+    (and (= networkState :error) (>= counter index) (< index 2))                   :error
+    :else                                                                           :pending))
+
+(defn progress-boxes
+  [networkState]
+  [rn/view
+   {:style style/progress-box-container}
+   (let [numbers (range 1 total-box)]
+     (doall (for [n numbers]
+              [progress-box/view
+               {:state    (calculate-box-state networkState (@app-state :counter) n)
+               :customization-color :blue
+                :key              n
+               }
+               ])))])
+
+(defn calculate-box-state-arbitrum
+  [networkState]
+  (cond
+    (= networkState :pending) :pending
+    (= networkState :error)   :error
+    :else                      :confirmed))
+
+(defn calculate-box-width
+  [showHalf]
+  (cond
+    (and showHalf (< (@app-state :counter) 30)) (- total-box (@app-state :counter))
+    showHalf                                    30
+    (< (@app-state :counter) total-box)         (- total-box (@app-state :counter))
+    :else                                       0))
+
+(defn progress-boxes-arbitrum
+  [networkState]
+  [rn/view
+   {:style style/progress-box-container}
+   [progress-box/view
+    {:state    (calculate-box-state-arbitrum networkState)
+    :customization-color :blue
+     }
+     ]
+   [rn/view
+    {:style            style/progress-box-arbitrum
+     :background-color (get-colors "neutral-5")
+     :border-color     (get-colors "neutral-10")}
+    [rn/view
+     (assoc
+      (let [box-style (cond
+                        (= networkState :finalising) (assoc {:style style/progress-box-arbitrum-abs}
+                                                             :right (str (calculate-box-width true) "%")
+                                                             :background-color
+                                                             (colors/custom-color-by-theme :blue 50 60))
+                        (= networkState :finalized)  (assoc {:style style/progress-box-arbitrum-abs}
+                                                             :right (str (calculate-box-width false) "%")
+                                                             :background-color
+                                                             (colors/custom-color-by-theme :blue 50 60))
```

Comment:
this: "(colors/custom-color-by-theme :blue 50 60)" is not how this function works ->
it should be:
```
(colors/theme-colors
(colors/custom-color customization-color 50)
(colors/custom-color customization-color 60)
theme)

```
where `theme` and `customization-color` are props that are passed in

Reply:
thanks , updated this now